### PR TITLE
Security upgrade rails from 7.0.8.1 to 7.1.0

### PR DESCRIPTION
**Pull request by deepak4566**
<p>This PR was created created with help of Snyk Scans .</p><br />                                                                              
<h3> This PR to fix one or more vulnerable packages in the `rubygems` dependencies of this project.</h3>

#### Changes included in this PR

- Changes to the following files to upgrade the vulnerable dependencies to a fixed version:
    - docs/Gemfile
    - docs/Gemfile.lock

##### With an upgrade:
Severity                   | Priority Score (*)                   | Issue                   | Breaking Change                   | Exploit Maturity
:-------------------------:|-------------------------|:-------------------------|:-------------------------|:-------------------------
medium severity  |  **479/1000**  <br/> **Why?** Has a fix available, CVSS 5.3  | Information Exposure <br/>[SNYK-RUBY-ACTIONCABLE-20338](https://snyk.io/vuln/SNYK-RUBY-ACTIONCABLE-20338) |  No  | No Known Exploit 
medium severity  |  **616/1000**  <br/> **Why?** Proof of Concept exploit, Has a fix available, CVSS 5.9  | Web Cache Poisoning <br/>[SNYK-RUBY-RACK-1061917](https://snyk.io/vuln/SNYK-RUBY-RACK-1061917) |  No  | Proof of Concept 

(*) Note that the real score may have changed since the PR was raised.

Check the changes in this PR to ensure they won't cause issues with your project.

------------

For more information:  
🧐 [View latest project report](https://app.snyk.io/org/22r21a12k0/project/c844aec0-ff61-4cf7-a607-acb207f8f739?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr)

🛠 [Adjust project settings](https://app.snyk.io/org/22r21a12k0/project/c844aec0-ff61-4cf7-a607-acb207f8f739?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr/settings)

[//]: # (snyk:metadata:{"prId":"56ab3c4c-c05c-4f8b-acaf-0cb3b180618a","prPublicId":"56ab3c4c-c05c-4f8b-acaf-0cb3b180618a","dependencies":[{"name":"rails","from":"7.0.8.1","to":"7.1.0"}],"packageManager":"rubygems","projectPublicId":"c844aec0-ff61-4cf7-a607-acb207f8f739","projectUrl":"https://app.snyk.io/org/22r21a12k0/project/c844aec0-ff61-4cf7-a607-acb207f8f739?utm_source=github&utm_medium=referral&page=fix-pr","type":"user-initiated","patch":[],"vulns":["SNYK-RUBY-ACTIONCABLE-20338","SNYK-RUBY-RACK-1061917"],"upgrade":["SNYK-RUBY-ACTIONCABLE-20338","SNYK-RUBY-RACK-1061917"],"isBreakingChange":false,"env":"prod","prType":"fix","templateVariants":["updated-fix-title","priorityScore"],"priorityScoreList":[479,616],"remediationStrategy":"vuln"})

---
Signed-off-by: Deepak [deepakreddyats1@gmail.com](mailto:epakreddyats1@gmail.com)

### CODE REVIEW

**Comment by deepak4566** [comment]
@MUzairS15 @MarioArriaga92  please take look on this pr , if this is ok , i may raise more pr's to solve vulnerabilities

**Comment by l5io** [comment]
🚀 Preview for commit bfdbc8895c4d5d76ed3a4fb8818a64fff74e74a1 at: https://65f576740252f19cd93de62b--meshery-docs-preview.netlify.app

**Review by MUzairS15 — COMMENTED** [review]
Please fix the DCO check.

File: docs/Gemfile [hunk 1]
```diff
@@ -25,7 +25,7 @@ gem 'jekyll-include-cache'
 group :graphqlDocs_plugins do
  gem "rake"
  gem "graphql"
- gem "rails"
+ gem "rails", ">= 7.1.0"
```
Comment (MUzairS15) [inline docs/Gemfile:28]: Should we pin to a specific version instead of >=

Comment (deepak4566) [reply docs/Gemfile:28]: yah we can "=" if we want exact or specific version

Comment (MUzairS15) [reply docs/Gemfile:28]: It's better to pin to a version,

Comment (deepak4566) [reply docs/Gemfile]: i just fixed it , but let me know about  tilde (~>) constraint.  This allows your project to use any version of Rails within the 7.1.x series, as long as it’s at least 7.1.0. ,  i see many projects use that

**Comment by l5io** [comment]
🚀 Preview for commit 0245e43a1a1c0c50ee5a0c4ce4f16ae3eccc9af2 at: https://65f80a18ba03423bd521c617--meshery-docs-preview.netlify.app

File: docs/Gemfile (hunk 1 above)
Comment (MUzairS15) [reply docs/Gemfile]: Yes, auto bump in minor version is ok.
Thanks.

**Comment by MUzairS15** [comment]
@MarioArriaga92

**Comment by MUzairS15** [comment]
DCO check @deepak4566

**Comment by deepak4566** [comment]
> DCO check @deepak4566

Can I know how to do DCO check for this PR because some of the commits were from snyk bot ??

**Comment by MarioArriaga92** [comment]
@deepak4566 @MUzairS15 just to clarify, I was expecting PR to exclusively include changes to existing ci yaml GitHub workflow (or new yaml file) so we could have snyk action scan for us on PRs/pushes:
https://github.com/snyk/actions

File: docs/Gemfile (hunk 1 above)
Comment (MarioArriaga92) [reply docs/Gemfile]: Should there be tests to check rails version is compatible with project libs?

Comment (MarioArriaga92) [reply docs/Gemfile]: “Check the changes in this PR to ensure they won't cause issues with your project.”

**Review by leecalcote — APPROVED** [review]

**Comment by deepak4566** [comment]
> @deepak4566 @MUzairS15 just to clarify, I was expecting PR to exclusively include changes to existing ci yaml GitHub workflow (or new yaml file) so we could have snyk action scan for us on PRs/pushes: https://github.com/snyk/actions

Yah @MarioArriaga92 I also wanted to have as as workflow of  snyk  and  I was getting some permission error to raise a integration pr , so that iam pushing fixes from my forked copy tho .synk free plan may work for us . Let me see about this

**Comment by deepak4566** [comment]
> > @deepak4566 @MUzairS15 just to clarify, I was expecting PR to exclusively include changes to existing ci yaml GitHub workflow (or new yaml file) so we could have snyk action scan for us on PRs/pushes: https://github.com/snyk/actions
> 
> Yah @MarioArriaga92 I also wanted to have as as workflow of snyk and I was getting some permission error to raise a integration pr , so that iam pushing fixes from my forked copy tho .synk free plan may work for us . But the main problem is only certain rangee of vulnerabilities can be fixable and remaining may not be . Those fixable vulnerabilities can be fixed by 10 to 15 pr's .Yah for the long term usage I think we need to have any of the SAST or SCA tool .

**Comment by leecalcote** [comment]
Site failing to build? Commits need to be signed off, @deepak4566.

@MarioArriaga92, you're right that we did. It generally didn't work and caused more pain that it was worth. Perhaps, it has improved since that time, though.

**Comment by MUzairS15** [comment]
I hope you can see the logs yourself by clicking on the `Details` for Netlify CI check. Deployment is failing.

**Comment by deepak4566** [comment]
@MUzairS15 @MarioArriaga92 @leecalcote  iam thinking to create branch called vuln-fixes in meshery , so that master may not effect with that and more over we can reduce the pain points , after we know fixes are correct ,we can just update branch , let me know so that , i can close this pr and open another one

**Comment by leecalcote** [comment]
Thanks, but my preference is to stay away from long-lived branches. When a dependency is updated or vulnerability addressed, this can be done in any branch created for that specific purpose (just like this PR).

**Comment by leecalcote** [comment]
Do sign-off on your commits, if you would, @deepak4566 - https://github.com/meshery/meshery/pull/10577/checks?check_run_id=22775343722

**Review by leecalcote — CHANGES_REQUESTED** [review]
Do sign-off on your commits, if you would, @deepak4566 - https://github.com/meshery/meshery/pull/10577/checks?check_run_id=22775343722

**Comment by l5io** [comment]
🚀 Preview for commit 5caea8169896acfa32d04af3da709b372b4fd66b at: https://65fd13ae2fcc07d3a1fb6bbf--meshery-docs-preview.netlify.app

**Comment by leecalcote** [comment]
@deepak4566 it'll take more than an empty commit.

**Comment by deepak4566** [comment]
> @deepak4566 it'll take more than an empty commit.

Yah im trying to signoff  though , but the problem is that snyk bot has commits where snyk is author , and not getting idea how do i sign off for a bot 
snyk_bot

**Comment by MarioArriaga92** [comment]
> > @deepak4566 it'll take more than an empty commit.
> 
> Yah im trying to signoff though , but the problem is that snyk bot has commits where snyk is author , and not getting idea how do i sign off for a bot , i think github supports auto signoff for bots?? snyk_bot

You should revert your recent commits locally, commit all changes in single commit with signoff and force push your changes so remote branch es overwritten accordingly @deepak4566

**Comment by MarioArriaga92** [comment]
> > > @deepak4566 it'll take more than an empty commit.
> > 
> > 
> > Yah im trying to signoff though , but the problem is that snyk bot has commits where snyk is author , and not getting idea how do i sign off for a bot , i think github supports auto signoff for bots?? snyk_bot
> 
> You should revert your recent commits locally, commit all changes in single commit with signoff and force push your changes so remote branch es overwritten accordingly @deepak4566

I meant “reset” local commits

**Comment by l5io** [comment]
🚀 Preview for commit 3652ee7b19f1d5db4a76773361e4ecc204c82092 at: https://65ffe28b011ef5a7e4473345--meshery-docs-preview.netlify.app

**Comment by deepak4566** [comment]
> > > @deepak4566 it'll take more than an empty commit.
> > 
> > 
> > Yah im trying to signoff though , but the problem is that snyk bot has commits where snyk is author , and not getting idea how do i sign off for a bot , i think github supports auto signoff for bots?? snyk_bot
> 
> You should revert your recent commits locally, commit all changes in single commit with signoff and force push your changes so remote branch es overwritten accordingly @deepak4566

Thanks @MarioArriaga92

**Comment by l5io** [comment]
🚀 Preview for commit 92dd667cea6f7d2297743b9879f89ab3886a1849 at: https://65ffe844622a35b29b3fc6e7--meshery-docs-preview.netlify.app

**Review by leecalcote — CHANGES_REQUESTED** [review]
Commits are signed off. The docs are failing to build, however.

**Comment by deepak4566** [comment]
> Commits are signed off. The docs are failing to build, however.

Yah , im looking into it

**Comment by deepak4566** [comment]
> Commits are signed off. The docs are failing to build, however.
  
I think docs are failing due to ruby-3.2.2 is required when upgrading  , i see that's a issue

**Comment by leecalcote** [comment]
> > Commits are signed off. The docs are failing to build, however.
> 
> I think docs are failing due to ruby-3.2.2 is required when upgrading , i see that's the issue

Ok, it's good to have the issue identified. What is being done about it? We're unable to merge this PR, if it's going to break the docs build.

**Comment by l5io** [comment]
🚀 Preview for commit 1ae2d84e14e08d094de5064a07a9e62acb11883c at: https://6602cd6c0cdeb133842671b1--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 469cc9a2b98e7ab2a3306a6a2ddb4affd3465f7e at: https://6602cf7b75754b35ef796581--meshery-docs-preview.netlify.app

**Comment by leecalcote** [comment]
Resyncing with master.

**Comment by l5io** [comment]
🚀 Preview for commit c73e1d683b45ad62e59b90913225cdf87f3bf6ce at: https://6619ce13bbe1231bf3fcba9d--meshery-docs-preview.netlify.app

**Comment by l5io** [comment]
🚀 Preview for commit 5e44b612fa97dd4c841c6f398eded09b0e1214cd at: https://6620e279c18fff2bc274b326--meshery-docs-preview.netlify.app

**Pull request closed without merging**